clevertap-core/src/test/java/com/clevertap/android/sdk/utils/CTCachesTest.kt

```diff
@@ -1,60 +1,72 @@
 package com.clevertap.android.sdk.utils
 
-import com.clevertap.android.sdk.TestLogger
+import android.graphics.Bitmap
+import com.clevertap.android.sdk.inapp.images.memory.Memory
+import io.mockk.every
+import io.mockk.mockk
 import org.junit.After
-import org.junit.Ignore
+import org.junit.Assert.assertNotSame
+import org.junit.Assert.assertSame
 import org.junit.Test
+import java.io.File
 import kotlin.test.assertEquals
 
-@Ignore
 class CTCachesTest {
+    @After
+    fun tearDown() {
+        CTCaches.clear() // Clear the singleton instance after each test
+    }
 
-    /*private val config1: CTCachesConfig = CTCachesConfig(
-            minImageCacheKb = 10,
-            minGifCacheKb = 10,
-            optimistic = 20,
-            maxImageSizeDiskKb = 2
-    )
+    @Test
+    fun `instance returns the same instance on multiple calls`() {
+        val mockImageMemory = mockk<Memory<Bitmap>>()
+        val mockGifMemory = mockk<Memory<ByteArray>>()
+        val mockFileMemory = mockk<Memory<ByteArray>>()
 
-    private val config2: CTCachesConfig = CTCachesConfig(
-            minImageCacheKb = 100,
-            minGifCacheKb = 100,
-            optimistic = 20,
-            maxImageSizeDiskKb = 2
-    )
+        val instance1 = CTCaches.instance(mockImageMemory, mockGifMemory, mockFileMemory)
+        val instance2 = CTCaches.instance(mockImageMemory, mockGifMemory, mockFileMemory)
+        assertSame(instance1, instance2)
```

Comment:
this can be more relevant test if we try on n threads in parallel and check in another test case, but i think we can avoid testing singleton pattern to work properly in general.